addOns/exportreport/src/main/java/org/zaproxy/zap/extension/exportreport/export/ReportExport.java

```diff
@@ -141,11 +148,61 @@ public static String generateDUMP(
                 Alert.RISK_INFO,
                 Constant.messages.getString("exportreport.risk.severity.info.label"));
 
-        List<Alert> alerts =
-                Control.getSingleton()
-                        .getExtensionLoader()
-                        .getExtension(ExtensionAlert.class)
-                        .getAllAlerts();
+        List<Alert> alerts = new ArrayList<>();
+
+        // check if scanid is valid when specified
+        ActiveScan scan = null;
+        if (scanId != -1) {
+            scan =
+                    Control.getSingleton()
+                            .getExtensionLoader()
+                            .getExtension(ExtensionActiveScan.class)
+                            .getScan(scanId);
+
+            if (scan == null) {
+                logger.warn(
```

Comment:
IMO we should throw an exception (e.g. `IllegalStateException`) instead, to ensure we return only the data requested.